tools/bin/go_core_tests_integration

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env bash
```

Comment:
We're at 4 scripts that share a lot of similar logic now, I think we should either consolidate them or export some helpers, otherwise this is going to be a pain to maintain. ChatGPT is usually quite good at working with shell scripts if you provide all 4 as context

Reply:
I can create a follow-up for this?

Testing this, especially the caching is a massive pain. I know this works, and would be good to have a logical starting point.

Reply:
Sure